noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/WebhoseCollector.java

```diff
@@ -0,0 +1,201 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.google.api.client.http.*;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.JsonObjectParser;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import org.apache.commons.io.FileUtils;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URLEncodedUtils;
+import org.kohsuke.args4j.CmdLineException;
+import org.kohsuke.args4j.CmdLineParser;
+import org.kohsuke.args4j.Option;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Properties;
+
+
+public class WebhoseCollector {
+    @Option(required = true,
+            name = "-a",
+            aliases = "--api",
+            usage = "Webhose API")
+    private String apiKey;
+
+    @Option(required = true,
+            name = "-o",
+            aliases = "--output",
+            usage = "output file directory")
+    private String filePath;
+
+    @Option(required = true,
+            name = "-t",
+            aliases = "--timestamp",
+            usage="timestamp for search")
+    private String ts;
+//    private static String propFileName = "";
+
+    public static void main( String[] args ) throws IOException {
+        new WebhoseCollector().doMain(args);
+    }
+
+    public void doMain(String[] args) throws IOException {
+        CmdLineParser parser = new CmdLineParser(this);
+        try{
+            parser.parseArgument(args);
+        } catch( CmdLineException e ) {
+            // if there's a problem in the command line,
+            // you'll get this exception. this will report
+            // an error message.
+            System.err.println(e.getMessage());
+            return;
+        }
+//        loadPropValue();
+
+        WebhoseQuery query = new WebhoseQuery();
+        query.title = "Zika";
+        query.siteTypes.add(WebhoseQuery.SiteType.news);
+
+        WebhoseResponse response = searchTimestamp(apiKey, query.toString(), Long.parseLong(ts));
+        String response_str = "";
+        if(response.totalResults == 0) {
+            System.err.println("No new result available.");
+            return;
+        }
+        while (true){
+
+            response_str += responseToString(response) + "\n";
+
+            if(response.moreResultsAvailable != 0) {
+                response = getMore(response);
+            } else {
+                //Update the timestamp
+                List<NameValuePair> params =
+                        URLEncodedUtils.parse(response.next, StandardCharsets.UTF_8);
+                for (final NameValuePair param : params) {
+                    if (param.getName().equals("ts")) {
+                        ts = param.getValue();
+                    }
+                }
+
+                File outFile = new File(filePath + "/response_" + ts + ".json");
```

Comment:
Could we rename the outfile's name to `response_from_to.json`? The current one is only record the end ts right?